src/std/net/httpd/handler.ss

```diff
@@ -598,47 +602,76 @@ END-C
-           (409 "Conflict")
-           (410 "Gone")
-           (411 "Length Required")
-           (412 "Precondition Failed")
-           (413 "Request Entity Too Large")
-           (414 "Request-URI Too Long")
-           (415 "Unsupported Media Type")
-           (416 "Requested Range Not Satisfiable")
-           (417 "Expectation Failed")
-           (500 "Internal Server Error")
-           (501 "Not Implemented")
-           (502 "Bad Gateway")
-           (503 "Service Unavailable")
-           (504 "Gateway Timeout")
-           (505 "HTTP Version Not Supported")))
+(def +http-response-codes+ (make-hash-table-eq))
+(defclass http-condition ())
+(defmethod {http-message http-condition}
+  (lambda (self) (hash-get +http-response-codes+ {http-code self})))
+(defrule (def-http-condition ctx number message)
+  (with-id ctx ((condition
+                 (string-substitute-char #\- #\space (stringify #'message))))
+    (defclass (condition http-condition) ())
+    (export condition)
+    (defmethod {http-code condition} (lambda (_) number))
+    (hash-put! +http-response-codes+ number message)))
+(defrules def-http-conditions ()
+  ((ctx (number message) ...) (begin (def-http-condition ctx number message) ...)))
+(def-http-conditions
+  (100 "Continue")
+  (101 "Switching Protocols")
+  (200 "OK")
+  (201 "Created")
+  (202 "Accepted")
+  (203 "Non-Authoritative Information")
+  (204 "No Content")
+  (205 "Reset Content")
+  (206 "Partial Content")
+  (300 "Multiple Choices")
+  (301 "Moved Permanently")
+  (302 "Found")
+  (303 "See Other")
+  (304 "Not Modified")
+  (305 "Use Proxy")
+  (307 "Temporary Redirect")
+  (400 "Bad Request")
+  (401 "Unauthorized")
+  (402 "Payment Required")
+  (403 "Forbidden")
+  (404 "Not Found")
+  (405 "Method Not Allowed")
+  (406 "Not Acceptable")
+  (407 "Proxy Authentication Required")
+  (408 "Request Timeout")
+  (409 "Conflict")
+  (410 "Gone")
+  (411 "Length Required")
+  (412 "Precondition Failed")
+  (413 "Request Entity Too Large")
+  (414 "Request-URI Too Long")
+  (415 "Unsupported Media Type")
+  (416 "Requested Range Not Satisfiable")
+  (417 "Expectation Failed")
+  (500 "Internal Server Error")
+  (501 "Not Implemented")
+  (502 "Bad Gateway")
+  (503 "Service Unavailable")
+  (504 "Gateway Timeout")
+  (505 "HTTP Version Not Supported"))
+
+(def (http-response-write-condition
+      res (condition #f) code: (code #f) content-type: (content-type #f) message: (message #f))
+  (http-response-write res (or code {http-code condition})
+                       `(("Content-Type" . ,(or content-type "text/plain")))
+                       (or message {http-message condition})))
+
+(def (condition-handler handler)
+  (lambda (req res)
+    (try (handler req res)
+     (catch (e)
+       (apply http-response-write-condition res
+              (if (http-condition? e)
+                [e]
+                [(Internal-Server-Error)]
+                #;[code: 500 message: "Internal Server Error"]))))))
```

Comment:
try to avoid apply and make a direct call where. 
Also what's the deal with the commented code? Either remove it or use it, no cruft please.